src/Workspaces/Core/Portable/CodeStyle/CodeStyleOptions.cs

```diff
@@ -134,6 +134,41 @@ public class CodeStyleOptions
                 EditorConfigStorageLocation.ForBoolCodeStyleOption("dotnet_style_explicit_tuple_names"),
                 new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.PreferExplicitTupleNames") });
 
+        private static readonly CodeStyleOption<NamedArgumentsPreference> s_NamedArgumentsPreferenceDefault =
+            new CodeStyleOption<NamedArgumentsPreference>(NamedArgumentsPreference.Literals, NotificationOption.None);
+
+        internal static readonly PerLanguageOption<CodeStyleOption<NamedArgumentsPreference>> PreferNamedArguments =
+            new PerLanguageOption<CodeStyleOption<NamedArgumentsPreference>>(
+                nameof(CodeStyleOptions), nameof(PreferNamedArguments), defaultValue: s_NamedArgumentsPreferenceDefault,
+                storageLocations: new OptionStorageLocation[] {
+                    new EditorConfigStorageLocation<CodeStyleOption<NamedArgumentsPreference>>("dotnet_style_prefer_named_arguments", s => ParseNamedArgumentsPreference(s)),
+                    new RoamingProfileStorageLocation($"TextEditor.%LANGUAGE%.Specific.{nameof(PreferNamedArguments)}")});
+
+        private static Optional<CodeStyleOption<NamedArgumentsPreference>> ParseNamedArgumentsPreference(string optionString)
+        {
+            if (TryGetCodeStyleValueAndOptionalNotification(optionString, out var value, out var notificationOpt))
+            {
+                if (value == "never")
+                {
+                    // If they provide 'never', they don't need a notification level.
+                    notificationOpt = notificationOpt ?? NotificationOption.None;
+                }
+
+                if (notificationOpt != null)
+                {
+                    switch (value)
+                    {
+                        case "never":
+                            return new CodeStyleOption<NamedArgumentsPreference>(NamedArgumentsPreference.Never, notificationOpt);
+                        case "for_literals":
+                            return new CodeStyleOption<NamedArgumentsPreference>(NamedArgumentsPreference.Literals, notificationOpt);
+                    }
+                }
+            }
+
+            return s_NamedArgumentsPreferenceDefault;
+        }
```

Comment:
I'm not very familiar with code style options. Is the intention that we could allow a list? For example `for_literals:warning,for_lambdas:error`. How would that be represented in the UI?
I'm wondering if we should have different options instead `prefer_named_arguments_for_literals` and `prefer_named_arguments_for_lambdas`.